plugins/modules/proxmox_kvm.py

```diff
@@ -1256,6 +1283,18 @@ def main():
         except Exception as e:
             module.fail_json(vmid=vmid, msg='Unable to revert settings on VM {0} with vmid {1}: Maybe is not a pending task...   '.format(name, vmid) + str(e))
 
+    if migrate:
```

Comment:
I find it a bit strange that this overrides `state=absent`, but then the other options above do it the same way.

Reply:
Well as it is, `migrate` would fail on the `proxmox.get_vm(vmid)` if it does not exist in the cluster, but indeed it will never ensure states when `migrate` (or previous actions) is `True`.  The way I see it, users need to define several tasks: one for managing the state and one or more for specific actions (`clone`, `delete`, `revert` and `migrate`).

Reply:
Nevermind, I just re-read this part and `clone` does not prevent state management. We _could_ do the same for `migrate`, but having `state` and `migrate` as mutually exclusive options seems easier to implement.